spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -99,11 +107,24 @@ module.exports = function (provider, setup) {
       });
   });
 
-  it("Pause and Resume work", function (done) {
+  it("Closes socket and open new one on the same port", function (done) {
+    socket.listen('127.0.0.1', 9981, function () {
+      socket.close(function () {
+        socket.listen('127.0.0.1', 9981, function () {
+          expect(true).toEqual(true);
+          PromiseCompat.all([
+            socket.close(function () {}),
+            done()]);
+        });
+      });
+    });
+  });
+
+  it("Pauses and resumes", function (done) {
     socket.connect('www.google.com', 80, function () {
```

Comment:
I think @bemasc wrote this test so he could speak for it better, but it looks like the GET request (https://github.com/freedomjs/freedom/blob/soycode-tcpintegration/spec/providers/coreIntegration/tcpsocket.integration.src.js#L150) happens in the callback for pause and so things should trigger in the right order.